perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserVirtualAttributeCollectedFromUserExtSource.java

```diff
@@ -157,39 +160,66 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 	}
 
 	/**
-	 * Get list of message patterns used for resolving attribute value change.
-	 * @return List of Patterns
+	 * Functional interface controlling AuditEvents
 	 */
-	public List<Pattern> getPatternsForMatch() {
-		List<Pattern> patterns = new ArrayList<>();
-		patterns.add(allAttributesRemovedForUserExtSource);
-		patterns.add(setUserExtSourceAttribute);
-		patterns.add(removeUserExtSourceAttribute);
+	@FunctionalInterface
+	public interface AttributeHandleIdenfier {
+		Integer check(AuditEvent ae);
+	}
+
 
-		return patterns;
+	/**
+	 * Returns userId if AuditEvent is correct and null if not.
```

Comment:
You don't need to use javadoc in method implementations, only in theri declarations -> in the functional interface.